java/src/main/java/bracketValidation/bracketValidation.java

```diff
@@ -0,0 +1,27 @@
+package bracketValidation;
+
+import java.util.HashSet;
+
+public class bracketValidation {
+    public static void main(String args[]) {
+        bracketValidation("()");
+    }
+
+    public static boolean bracketValidation(String input) {
+        HashSet<Character> h = new HashSet<>();

```

Comment:
Use a stack instead of a hashset. This will fix some of the weird bugs you've got.